lapis/job_io/htcondor.py

```diff
@@ -31,36 +33,54 @@ def htcondor_job_reader(
         "DiskUsage_RAW": 1.024 / 1024 / 1024,
     },
 ):
-    htcondor_reader = csv.DictReader(iterable, delimiter=" ", quotechar="'")
-
-    for row in htcondor_reader:
-        if float(row[used_resource_name_mapping["walltime"]]) <= 0:
+    input_file_type = iterable.name.split(".")[-1].lower()
+    if input_file_type == "json":
+        htcondor_reader = json.load(iterable)
+    elif input_file_type == "csv":
+        htcondor_reader = csv.DictReader(iterable, delimiter=" ", quotechar="'")
+    else:
+        logging.getLogger("implementation").error(
+            "Invalid input file %s. Job input file can not be read." % iterable.name
+        )
+    for entry in htcondor_reader:
+        if float(entry[used_resource_name_mapping["walltime"]]) <= 0:
             logging.getLogger("implementation").warning(
-                "removed job from htcondor import (%s)", row
+                "removed job from htcondor import (%s)", entry
             )
             continue
         resources = {}
         for key, original_key in resource_name_mapping.items():
             try:
-                resources[key] = float(row[original_key]) * unit_conversion_mapping.get(
-                    original_key, 1
-                )
+                resources[key] = float(
+                    entry[original_key]
+                ) * unit_conversion_mapping.get(original_key, 1)
             except ValueError:
                 pass
+
         used_resources = {
             "cores": (
-                (float(row["RemoteSysCpu"]) + float(row["RemoteUserCpu"]))
-                / float(row[used_resource_name_mapping["walltime"]])
+                (float(entry["RemoteSysCpu"]) + float(entry["RemoteUserCpu"]))
+                / float(entry[used_resource_name_mapping["walltime"]])
             )
             * unit_conversion_mapping.get(used_resource_name_mapping["cores"], 1)
         }
         for key in ["memory", "walltime", "disk"]:
             original_key = used_resource_name_mapping[key]
             used_resources[key] = float(
-                row[original_key]
+                entry[original_key]
             ) * unit_conversion_mapping.get(original_key, 1)
+
+        try:
+            resources["inputfiles"] = deepcopy(entry["Inputfiles"])
+            for filename, filespecs in resources["inputfiles"].items():
+                del filespecs["usedsize"]
```

Comment:
Be careful, here you might get an error that you don't expect. The key might not be included in your input file. We always have to be careful with user-facing interfaces. Users can give you anything, so we should protect ourselves :) 
I propose to guard this statement:

```suggestion
                if "usedsize" in filespecs:
                    del filespecs["usedsize"]
```